developer-workflow/c-api.rst

```diff
@@ -122,6 +122,28 @@ Guidelines for expanding/changing the public API
   - ``return 0``: lookup succeeded; no item was found
   - ``return 1``: lookup succeeded; item was found
 
+- APIs with output parameters should ensure that each output parameter has a
+  valid value, no matter the return value of the API. Example::
```

Comment:
It's unclear to me what "output parameters" stand for. I suppose that you mean output parameters passed as pointers such as ``int*``. And "valid" is unclear to me: do you mean "initialized"?

Instead of "no matter the return value", you may say: "in all cases (especially on error)".

Reply:
> It's unclear to me what "output parameters" stand for. I suppose that you mean output parameters passed as pointers such as ``int*``.

Yes. Do you have a suggestion for how to make it clearer?

> Instead of "no matter the return value", you may say: "in all cases (especially on error)".

Yes, that's clearer. Thanks!

Reply:
> And "valid" is unclear to me: do you mean "initialized"?

Yes. Invalid == random value == undefined behaviour.

Reply:
PTAL :)